providers/password.rb

```diff
@@ -19,9 +19,15 @@
 #
 require 'shellwords'
 
+
 action :set do
   r = execute "Assign mysql password for #{new_resource.user} user" do
-    query = "UPDATE user SET Password = PASSWORD('#{Shellwords.escape(new_resource.password)}') WHERE User = '#{Shellwords.escape(new_resource.user)}'"
+    # The Password column of table mysql.user was removed in MySQL 5.7. The used column is now authentication_string
+    pw_column_name = "Password"
+    if (node["platform"] == "ubuntu" and node["platform_version"].to_f >= 16.04) then pw_column_name = "authentication_string" end
```

Comment:
I agree, we should drop the condition and completely switch to authentication_string.